tsdb/store.go

```diff
@@ -651,6 +661,7 @@ func (s *Store) CreateShard(database, retentionPolicy string, shardID uint64, en
 	path := filepath.Join(s.path, database, retentionPolicy, strconv.FormatUint(shardID, 10))
 	shard := NewShard(shardID, path, walPath, sfile, opt)
 	shard.WithLogger(s.baseLogger)
+	shard.WithMetrics(s.metrics)
```

Comment:
I'm uneasy about this touch-point. In the beta.8 code, the storage architecture had been refactored so there was a single top-level TSI `Index`. In this implementation there's an `Index` per-shard, with potentially many shards. We don't do anything here to distinguish different shards in the metrics labels, so they'll all end up fusing together. Should we inject a label here [like we do for index partitions](https://github.com/influxdata/influxdb/pull/19912/files#diff-18f80f7e0688cd2a4a53981f14766c61b98bfb8a76bc0170cba27f96f6d49a8bR285-R288)?